s3datarepository/users.go

```diff
@@ -0,0 +1,447 @@
+package s3datarepository
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/YaleSpinup/ds-api/apierror"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func (s *S3Repository) ListUsers(ctx context.Context, id string) (map[string]interface{}, error) {
+	if id == "" {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", errors.New("empty id"))
+	}
+
+	name := id
+	if s.NamePrefix != "" {
+		name = s.NamePrefix + "-" + name
+	}
+
+	log.Infof("listing users of the s3datarepository %s", name)
+
+	groupName := fmt.Sprintf("%s-DsTmpGrp", name)
+	usersOutput, err := s.listGroupsUsers(ctx, groupName)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Debugf("got iam users response %+v", usersOutput)
+
+	output := make(map[string]interface{}, len(usersOutput))
+	for _, u := range usersOutput {
+		keyOut, err := s.IAM.ListAccessKeysWithContext(ctx, &iam.ListAccessKeysInput{UserName: u.UserName})
+		if err != nil {
+			return nil, ErrCode("failed to getting access keys for user "+aws.StringValue(u.UserName), err)
+		}
+
+		keys := make(map[string]string, len(keyOut.AccessKeyMetadata))
+		for _, k := range keyOut.AccessKeyMetadata {
+			keys[aws.StringValue(k.AccessKeyId)] = aws.StringValue(k.Status)
+		}
+
+		output[*u.UserName] = struct {
```

Comment:
Is it better to use `aws.StringValue` here? Only because we do that above.

Reply:
oh yeah - im not sure where that came from actually 👍